internal/clients/http_client/http_client.go

```diff
@@ -241,3 +241,12 @@ func NewHTTPClientForAccount(account *accounts.Account, timeout time.Duration, l
 		Transport: authTransport,
 	}, nil
 }
+
+func IsHTTPAgentErrorStatusOf(err error, status int) (*types.AgentError, bool) {
```

Comment:
Handy func to check if an error that is bubbling up is an AgentError of certain http status